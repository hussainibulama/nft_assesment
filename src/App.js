import './App.css';
import {Routes, Route} from 'react-router-dom';
import AllRoutes from './Components/Routes';
function App() {
  return (
    <div data-testid='app'>
      <Routes>
        {AllRoutes.map((item) => (
          <Route key={item.index} path={item.path} element={<item.element />} />
        ))}
      </Routes>
    </div>
  );
}

export default App;
