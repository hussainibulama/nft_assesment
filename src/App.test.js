import renderer from 'react-test-renderer';
import App from './App';
import {BrowserRouter} from 'react-router-dom';
import {render, screen} from '@testing-library/react';
describe('App.js Snapshot test', () => {
  it('should render Page', () => {
    render(<App />, {wrapper: BrowserRouter});
    const appPage = screen.getByTestId('app');
    expect(appPage).toBeInTheDocument();
  });
  it('should maintain App file snapsjot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
