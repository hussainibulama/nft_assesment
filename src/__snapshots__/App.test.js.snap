// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js Snapshot test should maintain App file snapsjot 1`] = `
<div
  data-testid="app"
>
  <div
    className="container_login"
  >
    <div
      className="login_box"
    >
      <div
        className="logo_area"
      >
        Weather Inc
      </div>
      <h3
        data-testid="Login"
      >
        Login
      </h3>
      <p
        className="header"
        data-testid="welcome-notes"
      >
        Please enter your login information to proceed
      </p>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form_separator"
        />
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1ow8cur-MuiFormControl-root-MuiTextField-root"
          data-testid="username"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="username"
            id="username-label"
          >
            Username
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="username"
              name="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Username
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="form_separator"
        />
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1ow8cur-MuiFormControl-root-MuiTextField-root"
          data-testid="password"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="password"
            id="password-label"
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="password"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="form_separator"
        />
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-1l02j10-MuiButtonBase-root-MuiButton-root"
          data-testid="submit-button"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
    <footer>
      <p
        data-testid="copyright"
      >
        Â© 2020 - 2023 Weather inc
      </p>
    </footer>
  </div>
</div>
`;
